src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1031,13 +1048,52 @@ private async Task CommonStreaming(ConnectionState connectionState, string strea
         return await invocationTask.ConfigureAwait(false);
     }
 
+    private Activity? StartActivity(ConnectionContext connection, string methodName)
+    {
+        var connectionUri = (connection.RemoteEndPoint is UriEndPoint e) ? e.Uri : null;
+        var serviceName = connectionUri?.AbsolutePath.TrimStart('/').TrimEnd('/');
```

Comment:
This info is constant for the lifetime of an active connection, should we just set it in `StartAsyncCore`?

Reply:
Yes, that can be done.

I asked some questions about service name [here](https://github.com/dotnet/aspnetcore/pull/57101#issuecomment-2260489868). What do you think?